web/src/components/DandisetsPage.vue

```diff
@@ -103,120 +109,146 @@ export default {
+          params: {
+            limit: DANDISETS_PER_PAGE,
+            offset: (page.value - 1) * DANDISETS_PER_PAGE,
+            sort: sortField.value,
+            sortdir: sortDir.value,
+            search: route.query.search,
+          },
+        });
+
+        girderDandisetRequest.value = {
+          data, headers, status, statusText,
+        };
       }
-      return this.girderDandisetRequest?.data || [];
-    },
-    totalDandisets() {
+    });
+
+    // Unified Django + Girder
+
+    const dandisets = computed(() => {
       if (toggles.DJANGO_API) {
-        return this.djangoDandisetRequest ? this.djangoDandisetRequest.data.count : 0;
+        return mostRecentDandisetVersions.value || [];
       }
-      return this.girderDandisetRequest ? this.girderDandisetRequest.headers['girder-total-count'] : 0;
-    },
-  },
-  asyncComputed: {
-    async girderDandisetRequest() {
+      return girderDandisetRequest.value?.data || [];
+    });
+
+    const pages = computed(() => {
+      let totalDandisets: number;
       if (toggles.DJANGO_API) {
-        return null;
+        totalDandisets = djangoDandisetRequest.value?.count || 0;
+      } else {
+        totalDandisets = girderDandisetRequest.value?.headers['girder-total-count'] || 0;
       }
-      const {
-        listingUrl, page, sortField, sortDir,
-        $route: {
-          query: {
-            search,
-          },
-        },
-      } = this;
-
-      const {
-        data, headers, status, statusText,
-      } = await girderRest.get(listingUrl, {
-        params: {
-          limit: DANDISETS_PER_PAGE,
-          offset: (page - 1) * DANDISETS_PER_PAGE,
-          sort: sortField,
-          sortdir: sortDir,
-          search,
-        },
-      });
+      return Math.ceil(totalDandisets / DANDISETS_PER_PAGE) || 1;
+    });
 
-      return {
-        data, headers, status, statusText,
-      };
-    },
-    async djangoDandisetRequest() {
-      if (!toggles.DJANGO_API) {
-        return null;
-      }
-      return publishRest.dandisets({ page: this.page, page_size: DANDISETS_PER_PAGE, user: this.user ? 'me' : null });
-    },
-    async mostRecentDandisetVersions() {
-      if (this.djangoDandisetRequest === null) {
-        return null;
-      }
-      return Promise.all(this.djangoDandisetRequest.data.results.map(
-        (dandiset) => publishRest.mostRecentVersion(dandiset.identifier),
-      ));
-    },
-  },
-  watch: {
-    queryParams(params) {
-      this.$router.replace({
-        ...this.$route,
+    const queryParams = computed(() => ({
+      page: String(page.value),
+      sortOption: String(sortOption.value),
+      sortDir: String(sortDir.value),
+    }));
+    watch(queryParams, (params) => {
+      ctx.root.$router.replace({
+        ...route,
+        // replace() takes a RawLocation, which has a name: string
+        // Route has a name: string | null, so we need to tweak this
+        name: route.name as string,
```

Comment:
Does coercing the type like this lead to trouble? If `route.name` is never expected to be `null`, can you instead perform the type narrowing above (before the invocation of `replace()`)?

Reply:
If we were to type narrow, we'd need to take an alternative action in the case where `route.name === null`. In the `RawLocation` type, `name` is optional. So while `null` isn't valid, we could cover our bases by instead writing:
```suggestion
        name: route.name || undefined,
```